pkg/asset/manifests/openstack/cloudproviderconfig.go

```diff
@@ -1,29 +1,25 @@
 package openstack
 
 import (
+	"io/ioutil"
 	"strconv"
 	"strings"
 
 	"github.com/gophercloud/utils/openstack/clientconfig"
+	"github.com/openshift/installer/pkg/asset/installconfig/openstack"
+	"github.com/openshift/installer/pkg/types"
 )
 
-// CloudProviderConfig generates the cloud provider config for the OpenStack platform.
-func CloudProviderConfig(cloud *clientconfig.Cloud) string {
-	res := `[Global]
-secret-name = openstack-credentials
-secret-namespace = kube-system
-`
-	if cloud.RegionName != "" {
-		res += "region = " + cloud.RegionName + "\n"
-	}
-
-	if cloud.CACertFile != "" {
-		res += "ca-file = /etc/kubernetes/static-pod-resources/configmaps/cloud-config/ca-bundle.pem\n"
-	}
-
-	return res
+// Error represents a failure while generating OpenStack provider
+// configuration.
+type Error struct {
```

Comment:
Why do we need this error wrapper? Can we use `Wrap` from `github.com/pkg/errors`?

Reply:
My rationale is twofold:
1. switching to the pure-stdlib `errors.Unwrapper` interface
2. removing one external dependency in a couple lines of code

Reply:
Happy to revert to Dave Cheney’s package if it serves consistency

Reply:
1) The error returned from `Wrap` already implements the `Unwrapper` interface.
2) The installer already has that dependency.
3) I could see that if it was a meaningful type that provided some more information about the type of the error or a way to get information specific to the error. I personally don't see that value in this case.

With that said, this is code that you are going to own. So if you feel that the value of the type outweighs the maintentance burden, then have at it.